snappi_ixnetwork/trafficitem.py

```diff
@@ -1277,48 +1293,97 @@ def results(self, request):
         ).Page
         if ixn_page.PageSize < flow_count:
             ixn_page.PageSize = flow_count
-        flow_stat = self._api.assistant.StatViewAssistant(
-            "Traffic Item Statistics"
-        )
-        for row in flow_stat.Rows:
-            name = row["Traffic Item"]
-            if len(flow_names) > 0 and name not in flow_names:
-                continue
-            if name in flow_rows:
-                flow_row = flow_rows[name]
+        if self._api._flow_tracking:
+            table = self._api.assistant.StatViewAssistant("Flow Statistics")
+            for row in table.Rows:
+                name = row["Traffic Item"]
                 if (
-                    float(row["Tx Frame Rate"]) > 0
-                    or int(row["Tx Frames"]) == 0
+                    len(flow_names) > 0
+                    and row["Traffic Item"] not in flow_names
```

Comment:
Why is the logical statement made to "AND" logic instead of "OR"?

Reply:
even though flow_names>0 condition satisfies, the `flow_name` requet by the user should be part of flow_names